consul/plan.sh

```diff
@@ -0,0 +1,39 @@
+pkg_origin=core
+pkg_name=consul
+pkg_version=0.7.0
+pkg_maintainer='The Habitat Maintainers <humans@habitat.sh>'
+pkg_license=("Apache 2")
+pkg_upstream=https://www.consul.io/
+pkg_source=https://releases.hashicorp.com/${pkg_name}/${pkg_version}/${pkg_name}_${pkg_version}_linux_amd64.zip
+pkg_shasum=b350591af10d7d23514ebaa0565638539900cdb3aaa048f077217c4c46653dd8
+pkg_filename=${pkg_name}-${pkg_version}.zip
+pkg_deps=()
+pkg_build_deps=(core/unzip core/wget)
+pkg_bin_dirs=(bin)
+pkg_include_dirs=(include)
+pkg_lib_dirs=(lib)
+pkg_expose=(8400 8500 8600)
+pkg_svc_user=root
+
+do_download() {
+  wget -O "${HAB_CACHE_SRC_PATH}"/"${pkg_filename}" ${pkg_source}
+}
+
+do_unpack() {
+  mkdir -p "${pkg_prefix}"
+  cp "${HAB_CACHE_SRC_PATH}"/"${pkg_filename}" "${pkg_prefix}"
+  cd "${pkg_prefix}" || exit
+  unzip ${pkg_filename}
+  rm ${pkg_filename}
+}
+
+do_build() {
+    return 0
+}
+
+do_install() {
+  cd "${pkg_prefix}" || exit
+  mkdir bin
+  mv consul bin/
+  chmod 755 "${pkg_prefix}"/bin/consul
+}
```

Comment:
Thanks to @fnichol, I have learned the power of `install` where the following is the equivalent to:
`mkdir -p "${pkg_prefix}"/bin && cp consul "${pkg_prefix}"/bin`

```shell
do_install() {
  install -D consul "${pkg_prefix}"/bin/consul
}
```